File: src/main/java/com/github/bjansen/gatling/blocking/javaapi/actions/BlockingActionBuilder.java
package com.github.bjansen.gatling.blocking.javaapi.actions;

import io.gatling.javaapi.core.ActionBuilder;

public class BlockingActionBuilder implements ActionBuilder {
	
	private final String description;
	private final Runnable runnable;

	public BlockingActionBuilder(String description, Runnable runnable) {
		this.description = description;
		this.runnable = runnable;
	}

	@Override
	public io.gatling.core.action.builder.ActionBuilder asScala() {
		return new com.github.bjansen.gatling.blocking.actions.BlockingActionBuilder(description, runnable);
	}
}

File: src/main/scala/com/github/bjansen/gatling/blocking/actions/BlockingActionBuilder.scala
package com.github.bjansen.gatling.blocking.actions

import io.gatling.core.action.Action
import io.gatling.core.action.builder.ActionBuilder
import io.gatling.core.structure.ScenarioContext

class BlockingActionBuilder(description: String, runnable: Runnable) extends ActionBuilder {

  override def build(ctx: ScenarioContext, next: Action): Action = {
    new BlockingCodeAction(description, runnable, ctx, next)
  }
}
